### PR TITLE
Speed curve example

### PR DESCRIPTION
**Pull request by devinaconley**
## Overview
This PR implements a target speed curve example with the PID controller. The example loads up a CSV file of time series speed data, and dynamically feeds these values to the PID controllers as the target speed. It also dumps results to file for further analysis. This addresses #107.

## Changes
- adds `speed_curve.cpp` example within the `lane_follower` example set
- adds mock speed curve data (generated programmatically)
- updates readme
- new solo straightaway scenario
- handles zero speed case

## Testing
Build and run the `speed_curve` example. Should see the vehicle smoothly accelerate up to 20 m/s before slowing back to a stop. There should also be an output file generated called `speed_curve_reproduced.csv`. This can be examined to ensure curves match relatively well.

### CODE REVIEW

**Comment by devinaconley** [comment]
Example comparison of mock vs reproduced data
speed_curve_comparison

File: examples/cpp/lane_follower/speed_curve.cpp [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+#include "sensor_config.h"
+#include "DataFrame.h"
+#include "Stopwatch.h"
+
+#include "opencv2/core.hpp"
+#include "opencv2/core/utility.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/imgproc.hpp"
+
+#include "eigen3/Eigen/Dense"
```
Comment (bradleynull) [inline examples/cpp/lane_follower/speed_curve.cpp:21]: This is the wrong include for Windows. Should be

```
#include <Eigen/Eigen>
```

**Review by bradleynull — CHANGES_REQUESTED** [review]
@devinaconley Just some minor cleanup and a bug fix for Windows. Everything seems to be working for me though.

File: examples/cpp/lane_follower/README.md [hunk 2]
```diff
@@ -46,6 +46,18 @@ Run example
 ```
 
 
+### Lane follower - speed curve
+This example uses the PID controller with a time series of target speeds to reproduce a speed curve.
+
+Currently the example uses the mock data found in `speed_curve_data.csv`. This can be replaced with your own
+data in order to compare or calibrate the vehicle physics model used in the simulator.
```
[the remark this answers is not in the document]
Comment (devinaconley) [reply examples/cpp/lane_follower/README.md:53]: added

File: examples/cpp/lane_follower/speed_curve.cpp [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+                          cv::Point(int(bbox.xmax), int(bbox.ymax)),
+                          cv::Scalar(0, 0, 255));
+    }
+    cv::imshow("monoDrive", img);
+    cv::waitKey(1);
+}
+
+int main(int argc, char** argv)
+{
+    MatrixXd data = read_csv<MatrixXd>("examples/cpp/lane_follower/speed_curve_data.csv");
```
[the remark this answers is not in the document]
Comment (devinaconley) [reply examples/cpp/lane_follower/speed_curve.cpp:171]: is that somewhat matlab centric?

File: examples/cpp/lane_follower/speed_curve.cpp (hunk 1 above)
Comment (devinaconley) [reply examples/cpp/lane_follower/speed_curve.cpp]: fixed :+1:

**Comment by devinaconley** [comment]
@bradleynull - ready for another look!

**Comment by devinaconley** [comment]
@bradleynull - addressed that last comment! Good another look.

**Review by bradleynull — APPROVED** [review]
@devinaconley Looks good to go. Approved!

**Pull request merged**